
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.2.3, mkdocs-material-8.1.9">
    
    
      
        <title>Downstream analysis - Primer to IMC analysis</title>
      
    
    
      <link rel="stylesheet" href="assets/stylesheets/main.2b4465f4.min.css">
      
        
        <link rel="stylesheet" href="assets/stylesheets/palette.e6a45f82.min.css">
        
          
          
          <meta name="theme-color" content="#009485">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="css/extra.css">
    
    <script>__md_scope=new URL(".",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="teal" data-md-color-accent="teal">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#downstream-analysis-of-imc-data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="index.html" title="Primer to IMC analysis" class="md-header__button md-logo" aria-label="Primer to IMC analysis" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Primer to IMC analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Downstream analysis
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/BodenmillerGroup/IMCWorkflow/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    BodenmillerGroup/IMCWorkflow
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="index.html" title="Primer to IMC analysis" class="md-nav__button md-logo" aria-label="Primer to IMC analysis" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    Primer to IMC analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/BodenmillerGroup/IMCWorkflow/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    BodenmillerGroup/IMCWorkflow
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="index.html" class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="intro.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="process.html" class="md-nav__link">
        Image processing
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="viewers.html" class="md-nav__link">
        Image visualization
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Downstream analysis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="analysis.html" class="md-nav__link md-nav__link--active">
        Downstream analysis
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#reading-in-the-data" class="md-nav__link">
    Reading in the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#spillover-correction" class="md-nav__link">
    Spillover correction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    Quality control
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cell-type-identification" class="md-nav__link">
    Cell-type identification
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-visualization" class="md-nav__link">
    Data visualization
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#spatial-analysis" class="md-nav__link">
    Spatial analysis
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#example-datasets" class="md-nav__link">
    Example datasets
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#reading-in-the-data" class="md-nav__link">
    Reading in the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#spillover-correction" class="md-nav__link">
    Spillover correction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    Quality control
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cell-type-identification" class="md-nav__link">
    Cell-type identification
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-visualization" class="md-nav__link">
    Data visualization
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#spatial-analysis" class="md-nav__link">
    Spatial analysis
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#example-datasets" class="md-nav__link">
    Example datasets
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
<a href="https://github.com/BodenmillerGroup/IMCWorkflow/edit/master/docs/analysis.md" title="Edit this page" class="md-content__button md-icon">
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
</a>


<h1 id="downstream-analysis-of-imc-data">Downstream analysis of IMC data</h1>
<p>This page gives a brief overview of the common steps for downstream analysis of IMC&trade; data.
For a more detailed workflow using example data, please refer to the <a href="https://github.com/BodenmillerGroup/IMCDataAnalysis">IMC Data Analysis</a> repository (under development).</p>
<p>The analysis described here relies on the <a href="https://www.bioconductor.org/">Bioconductor</a> framework for single-cell analysis. 
The book <a href="https://bioconductor.org/books/release/OSCA/">Orchestrating Single-Cell Analysis with Bioconductor</a> provides a good introduction into general single-cell analysis tasks.
Although the book mainly focuses on single-cell RNA sequencing data analysis, most concepts (e.g., data handling, clustering, and visualization) can be applied to single-cell IMC data.
The book is offered in <a href="https://bioconductor.org/books/3.14/OSCA.basic/">basic</a> and <a href="https://bioconductor.org/books/3.14/OSCA.advanced/">advanced</a> versions.</p>
<h2 id="reading-in-the-data">Reading in the data</h2>
<p>The <a href="https://github.com/BodenmillerGroup/imcRtools">imcRtools</a><sup id="fnref:fn1"><a class="footnote-ref" href="#fn:fn1">1</a></sup> package reads data generated by <code>steinbock</code> (using <a href="https://bodenmillergroup.github.io/imcRtools/reference/read_steinbock.html">read_steinbock</a>) or by the IMC Segmentation Pipeline (using <a href="https://bodenmillergroup.github.io/imcRtools/reference/read_cpout.html">read_cpout</a>) for downstream analysis in <a href="https://cran.r-project.org/">R</a>. 
Single-cell data (mean intensities per cell and channel, morphological features and locations, and spatial object graphs) are read into <a href="https://bioconductor.org/packages/release/bioc/html/SpatialExperiment.html">SpatialExperiment</a> or <a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html">SingleCellExperiment</a> objects.</p>
<p>The structure is as follows:</p>
<ul>
<li>mean intensity per cell and channel are stored in the <strong>counts</strong> assay of the objects (cells as columns, channels as rows).</li>
<li>cell-specific morphological features and image information are stored in the <strong>colData</strong> slot of the objects</li>
<li>marker-specific information extracted from the panel file are stored in the <strong>rowData</strong> slot of the objects</li>
<li>spatial object graphs are stored as edge lists in the <strong>colPair</strong> slot of the objects</li>
</ul>
<p>With the <code>SingleCellExperiment</code> object, the locations of cells are stored in the <code>colData</code> slot. For the <code>SpatialExperiment</code> container, the cells' locations are stored in the <code>spatialCoords</code> slot.</p>
<h2 id="spillover-correction">Spillover correction</h2>
<p>Channel-to-channel spillover that can occur between neighboring channels leads to false signals. This should be corrected as described previously<sup id="fnref:fn2"><a class="footnote-ref" href="#fn:fn2">2</a></sup>.
Spillover correction is an <strong>optional</strong> step in IMC data analysis and requires a specific experimental setup:</p>
<p>On an agarose slide, metal-tagged antibodies are spotted individually, and an individual region of interest is ablated for each antibody. 
Data from individual regions are processed using <a href="https://github.com/BodenmillerGroup/imcRtools">imcRtools</a> as explained in the <a href="https://bodenmillergroup.github.io/IMCDataAnalysis/spillover-correction.html">spillover correction</a> section of <a href="https://github.com/BodenmillerGroup/IMCDataAnalysis">IMC data analysis</a>. After obtaining a spillover matrix, single-cell data can be compensated using the <a href="https://bioconductor.org/packages/release/bioc/vignettes/CATALYST/inst/doc/preprocessing.html#compensation">CATALYST</a> package.</p>
<h2 id="quality-control">Quality control</h2>
<p>Quality control (QC) is a crucial step to avoid technical artefacts in downstream analyses. 
Broadly speaking, there are two levels of data quality control:</p>
<ol>
<li>
<p><strong>Pixel-level QC:</strong> The first step after image acquisition and pre-processing is the visual assessment of image quality. For this, interactive tools such as <a href="https://github.com/BodenmillerGroup/napari-imc">napari-imc</a>, <a href="https://github.com/BodenmillerGroup/histoCAT">histoCAT</a>, <a href="https://qupath.github.io/">QuPath</a>, or <a href="https://bioconductor.org/packages/release/bioc/html/cytomapper.html">cytomapper</a> (see below) can be used. Considerations when viewing images for the first time are: (i) are markers known to be specifically expressed detected as expected (e.g., high signal-to-noise ratio, cell type specificity, adequate maximum intensity per channel, etc.), (ii) is channel-spillover visually detectable, and (iii) are staining differences detectable between acquisitions? </p>
</li>
<li>
<p><strong>Cell-level QC:</strong> After reading in the single-cell data (e.g. using <code>imcRtools</code>), cell-level QC includes (i) global staining differences between acquisitions or batches (e.g., in form of ridge plots using <a href="https://www.bioconductor.org/packages//release/bioc/html/dittoSeq.html">dittoSeq</a>), (ii) marker-to-marker correlations across all cells, (iii) low-dimensional representations of the data (see below), and (iv) the cell density and cell size distributions per acquisition.</p>
</li>
</ol>
<h2 id="cell-type-identification">Cell-type identification</h2>
<p>One of the first analysis tasks includes cell phenotyping. 
This is commonly done by grouping cells based on marker expression and labeling these groups based on their biological roles.
There are three major approaches for identification of cell types in the single-cell IMC data.</p>
<ol>
<li>
<p><strong>Unsupervised clustering-based:</strong> Clustering is the most common approach for identification of cell types in an unbiased manner. To cluster single-cells based on their marker expression, software packages have been developed that utilize graph-based clustering strategies (e.g., <a href="https://github.com/JinmiaoChenLab/Rphenograph">Rphenograph</a>, <a href="https://bioconductor.org/packages/release/bioc/vignettes/scran/inst/doc/scran.html#5_Graph-based_clustering">scran</a>) or self-organising maps (the <code>flowSOM</code> implementation in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/CATALYST/inst/doc/differential.html#cluster-flowsom-clustering-consensusclusterplus-metaclustering">CATALYST</a> package). The <a href="https://www.bioconductor.org/packages/release/bioc/html/bluster.html">bluster</a> package offers a wide variety of clustering strategies applicable to single-cell data. After clustering, the mean marker expression per group is used to label clusters.</p>
</li>
<li>
<p><strong>Gating and classification:</strong> An alternative and more supervised approach to identify cell types from highly multiplexed images relies on manual gating (labeling) of cells and consecutive classification of all unlabeled cells. The <code>cytomapperShiny</code> application available in the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/cytomapper/inst/doc/cytomapper.html#12_Gating_cells_on_images">cytomapper</a> package allows to gate cells based on marker expression with joint visualization of the selected cells on composite images. The ground truth labels are saved and can be used as training dataset for general machine learning approaches (e.g., using <a href="https://topepo.github.io/caret/index.html">caret</a>, <a href="https://mlr.mlr-org.com/">mlr</a> or <a href="https://www.tidymodels.org/">tidymodels</a>). We observed a high classification accuracy when applying random forest-based classification. Additionally, a more specialized framework for classifying cells is the <a href="https://bioconductor.org/packages/release/bioc/html/SingleR.html">SingleR</a> package. </p>
</li>
<li>
<p><strong>Classification based on prior knowledge:</strong> Classification approaches that do not rely on manual labeling of cells were implemented in the R <a href="https://cole-trapnell-lab.github.io/garnett/docs/">Garnett</a> and Python <a href="https://astir.readthedocs.io/en/latest/index.html">Astir</a> modeling frameworks. Using these strategies, the user defines which cell types express which markers and cell labeling is performed automatically. </p>
</li>
</ol>
<h2 id="data-visualization">Data visualization</h2>
<p>Visualization for bioimaging data can be performed on the <strong>pixel-</strong> and <strong>single-cell</strong> levels:</p>
<p><strong>Pixel level visualization</strong></p>
<p>Interactive visualization can be performed using GUI-based tools as described in the <a href="viewers.html">Image visualization</a> section. 
As an alternative, the <a href="https://github.com/BodenmillerGroup/cytomapper">cytomapper</a> Bioconductor package supports handling and visualization of multi-channel images using the statistical programming language R. </p>
<p>The main visualization functionalities of <code>cytomapper</code> are three-fold:</p>
<p><em>plotPixels</em></p>
<p>The function takes a <code>CytoImageList</code> object (available via the <code>cytomapper</code> package) containing multi-channel images representing pixel-level expression values, and <em>optionally</em> a <code>CytoImageList</code> object containing segmentation masks and a <code>SingleCellExperiment</code> object containing cell-level metadata.</p>
<p>It allows the visualization of pixel-level information of up to six channels and outlining of cells based on cell-level metadata.</p>
<p><em>plotCells</em></p>
<p>This function takes a <code>CytoImageList</code> object containing segmentation masks, and a <code>SingleCellExperiment</code> object containing cell-level mean expression values and metadata information.</p>
<p>It allows the visualization of cell-level expression data and metadata information.</p>
<p><em>cytomapperShiny</em></p>
<p>This Shiny application allows gating of cells based on their expression values, and visualizes selected cells on their corresponding images. </p>
<p>It requires at least a <code>SingleCellExperiment</code> as input and optionally <code>CytoImageList</code> objects containing segmentation masks and multi-channel images.</p>
<p><strong>Single-cell level visualization</strong></p>
<p>Multiple Bioconductor packages have been released to visualize single-cell data contained in a <code>SingleCellExperiment</code> or a <code>SpatialExperiment</code> object.</p>
<p>A common approach to single-cell visualization is to first perform dimensionality reduction via PCA, tSNE, UMAP or Diffusion Maps. 
These functions are available in the <a href="https://bioconductor.org/packages/release/bioc/html/scater.html">scater</a> Bioconductor package. </p>
<p>The <code>scater</code> package further supports visualization of cells in low dimensions and a variety of functions to visualize single-cell expression (see <a href="https://bioconductor.org/packages/release/bioc/vignettes/scater/inst/doc/overview.html#3_Visualizing_expression_values">here</a>).</p>
<p>In addition, the <a href="https://www.bioconductor.org/packages//release/bioc/html/dittoSeq.html">dittoSeq</a> package offers an extensive range of visualization functionality using a <code>SingleCellExperiment</code> object as input.</p>
<p>For spatial visualization of single-cells and their interactions, the <code>imcRtools</code> package exports the <a href="https://bodenmillergroup.github.io/imcRtools/reference/plotSpatial.html">plotSpatial</a> function.</p>
<h2 id="spatial-analysis">Spatial analysis</h2>
<p>A number of data analysis approaches have previously been described to extract biological information from spatially annotated, single-cell resolved data<sup id="fnref:fn3"><a class="footnote-ref" href="#fn:fn3">3</a></sup><sup id="fnref:fn4"><a class="footnote-ref" href="#fn:fn4">4</a></sup><sup id="fnref:fn5"><a class="footnote-ref" href="#fn:fn5">5</a></sup><sup id="fnref:fn6"><a class="footnote-ref" href="#fn:fn6">6</a></sup><sup id="fnref:fn7"><a class="footnote-ref" href="#fn:fn7">7</a></sup>.
The <a href="https://github.com/BodenmillerGroup/imcRtools">imcRtools</a> package standardizes spatial data analysis by providing functionalities that wrap around the <code>SpatialExperiment</code> and <code>SingleCellExperiment</code> object classes. </p>
<p>Whereas <code>steinbock</code> and the IMC Segmentation Pipeline compute spatial object graphs, the <code>imcRtools</code> package provides the <a href="https://bodenmillergroup.github.io/imcRtools/reference/buildSpatialGraph.html">buildSpatialGraph</a> function for constructing interaction graphs "on the fly".  </p>
<p>The <a href="https://bodenmillergroup.github.io/imcRtools/reference/aggregateNeighbors.html">aggregateNeighbors</a> function computes the proportions of cell types within the direct neighborhood of each cell. Alternatively, expression counts of neighboring cells are summarized. These summarized values can be used to cluster cells and to detect cellular neighborhoods<sup id="fnref2:fn3"><a class="footnote-ref" href="#fn:fn3">3</a></sup><sup id="fnref2:fn6"><a class="footnote-ref" href="#fn:fn6">6</a></sup>.</p>
<p>The <a href="https://bodenmillergroup.github.io/imcRtools/reference/patchDetection.html">patchDetection</a> function detects spatial clusters of defined types of cells. By defining a certain distance threshold, all cells within the vicinity of these clusters are detected as well<sup id="fnref2:fn7"><a class="footnote-ref" href="#fn:fn7">7</a></sup>. </p>
<p>The <a href="https://bodenmillergroup.github.io/imcRtools/reference/countInteractions.html">countInteractions</a> and <a href="https://bodenmillergroup.github.io/imcRtools/reference/testInteractions.html">testInteractions</a> functions allow counting and testing the average cell type to cell type interactions per image<sup id="fnref2:fn4"><a class="footnote-ref" href="#fn:fn4">4</a></sup><sup id="fnref2:fn5"><a class="footnote-ref" href="#fn:fn5">5</a></sup>. </p>
<h2 id="example-datasets">Example datasets</h2>
<p>The <a href="https://github.com/BodenmillerGroup/imcdatasets">imcdatasets</a> Bioconductor package provides a collection of publicly available IMC datasets. 
These can be used for data re-analysis and methods development and are provided in standardized formats using the <code>SingleCellExperiment</code> and <code>CytoImageList</code> object classes.</p>
<div class="footnote">
<hr />
<ol>
<li id="fn:fn1">
<p>Windhager J. <em>et al.</em> (2021) An end-to-end workflow for multiplexed image processing and analysis. bioRxiv&#160;<a class="footnote-backref" href="#fnref:fn1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="fn:fn2">
<p>Chevrier S. <em>et al.</em> (2018) Compensation of Signal Spillover in Suspension and Imaging Mass Cytometry. Cell Systems&#160;<a class="footnote-backref" href="#fnref:fn2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="fn:fn3">
<p>Goltsev Y. <em>et al.</em> (2018) Deep Profiling of Mouse Splenic Architecture with CODEX Multiplexed Imaging. Cell 174&#160;<a class="footnote-backref" href="#fnref:fn3" title="Jump back to footnote 3 in the text">&#8617;</a><a class="footnote-backref" href="#fnref2:fn3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="fn:fn4">
<p>Schapiro D. <em>et al.</em> (2017) histoCAT: analysis of cell phenotypes and interactions in multiplex image cytometry data. Nature Methods&#160;<a class="footnote-backref" href="#fnref:fn4" title="Jump back to footnote 4 in the text">&#8617;</a><a class="footnote-backref" href="#fnref2:fn4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
<li id="fn:fn5">
<p>Schulz D. <em>et al.</em> (2018) Simultaneous Multiplexed Imaging of mRNA and Proteins with Subcellular Resolution in Breast Cancer Tissue Samples by Mass Cytometry. Cell Systems&#160;<a class="footnote-backref" href="#fnref:fn5" title="Jump back to footnote 5 in the text">&#8617;</a><a class="footnote-backref" href="#fnref2:fn5" title="Jump back to footnote 5 in the text">&#8617;</a></p>
</li>
<li id="fn:fn6">
<p>Sch√ºrch C. M. <em>et al.</em> (2021) Coordinated Cellular Neighborhoods Orchestrate Antitumoral Immunity at the Colorectal Cancer Invasive Front. Cell&#160;<a class="footnote-backref" href="#fnref:fn6" title="Jump back to footnote 6 in the text">&#8617;</a><a class="footnote-backref" href="#fnref2:fn6" title="Jump back to footnote 6 in the text">&#8617;</a></p>
</li>
<li id="fn:fn7">
<p>Hoch T. <em>et al.</em> (2021) Multiplexed Imaging Mass Cytometry of Chemokine Milieus in Metastatic Melanoma Characterizes Features of Response to Immunotherapy. bioRxiv&#160;<a class="footnote-backref" href="#fnref:fn7" title="Jump back to footnote 7 in the text">&#8617;</a><a class="footnote-backref" href="#fnref2:fn7" title="Jump back to footnote 7 in the text">&#8617;</a></p>
</li>
</ol>
</div>

              
            </article>
          </div>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="viewers.html" class="md-footer__link md-footer__link--prev" aria-label="Previous: Image visualization" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Image visualization
            </div>
          </div>
        </a>
      
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": ".", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "assets/javascripts/workers/search.22074ed6.min.js"}</script>
    
    
      <script src="assets/javascripts/bundle.960e086b.min.js"></script>
      
    
  </body>
</html>